src/neuroglancer/datasource/deepzoom/backend.ts

```diff
@@ -0,0 +1,114 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc., 2023 Gergely Csucs
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {WithParameters} from 'neuroglancer/chunk_manager/backend';
+import {WithSharedCredentialsProviderCounterpart} from 'neuroglancer/credentials_provider/shared_counterpart';
+import {ImageTileEncoding, ImageTileSourceParameters} from 'neuroglancer/datasource/deepzoom/base';
+import {VolumeChunk, VolumeChunkSource} from 'neuroglancer/sliceview/volume/backend';
+import {CancellationToken} from 'neuroglancer/util/cancellation';
+import {isNotFoundError, responseArrayBuffer} from 'neuroglancer/util/http_request';
+import {cancellableFetchSpecialOk, SpecialProtocolCredentials} from 'neuroglancer/util/special_protocol_request';
+import {registerSharedObject} from 'neuroglancer/worker_rpc';
+import {decodeJpeg} from 'src/neuroglancer/async_computation/decode_jpeg_request';
+import {decodePng} from 'src/neuroglancer/async_computation/decode_png_request';
+import {requestAsyncComputation} from 'src/neuroglancer/async_computation/request';
+import {transposeArray2d} from 'src/neuroglancer/util/array';
+
+/* This is enough if support for these aren't needed:
+ * - Firefox before 105 (OffscreenCanvas, 2022-09-20)
+ * - Safari before 16.4 (OffscreenCanvas, 2023-03-27)
+ */
+// declare var OffscreenCanvas: any; // shutting up some outdated compiler(?)
+
+@registerSharedObject() export class DeepzoomImageTileSource extends
+(WithParameters(WithSharedCredentialsProviderCounterpart<SpecialProtocolCredentials>()(VolumeChunkSource), ImageTileSourceParameters)) {
+  gridShape = (() => {
+    const gridShape = new Uint32Array(3);
+    const {upperVoxelBound, chunkDataSize} = this.spec;
+    for (let i = 0; i < 3; ++i) {
```

Comment:
Should go to `2` rather than `3`.

Reply:
Done in https://github.com/google/neuroglancer/pull/453/commits/a5737feb51489a7b612d8bee5a376612ef25b608
Also reduced the size of the array itself, it still seems to work.